### PR TITLE
Add Ukrainian translation file

### PR DESCRIPTION
**Pull request by kromanenko**
Please review new translation file

### CODE REVIEW

**Comment by MichaelPFrey** [comment]
Travis reports two errors:
```
msgfmt -c -v -o ./locale/ua/LC_MESSAGES/openscad.mo ./locale/ua.po
msgfmt: ./locale/ua.po: warning: PO file header fuzzy
                        warning: older versions of msgfmt will give an error on this
./locale/ua.po:9: invalid nplurals value
                  Try using the following, valid for Ukrainian:
                  "Plural-Forms: nplurals=3; plural=(n%10==1 && n%100!=11 ? 0 : n%10>=2 && n%10<=4 && (n%100<10 || n%100>=20) ? 1 : 2);\n"
./locale/ua.po:9: invalid plural expression
                  Try using the following, valid for Ukrainian:
                  "Plural-Forms: nplurals=3; plural=(n%10==1 && n%100!=11 ? 0 : n%10>=2 && n%10<=4 && (n%100<10 || n%100>=20) ? 1 : 2);\n"
msgfmt: found 2 fatal errors
316 translated messages, 2 untranslated messages.
error running msgfmt
make: *** [openscad] Error 1
```

File: locale/ua.po [hunk 1]
```diff
@@ -0,0 +1,1446 @@
...
+"Report-Msgid-Bugs-To: \n"
+"POT-Creation-Date: 2018-08-15 15:07+0200\n"
+"PO-Revision-Date: YEAR-MO-DA HO:MI+ZONE\n"
+"Last-Translator: Kyrylo Romanenko <romanenko-kv@hotmail.com>\n"
+"Language-Team: Ukrainian\n"
+"Language: ua\n"
+"MIME-Version: 1.0\n"
+"Content-Type: text/plain; charset=UTF-8\n"
+"Content-Transfer-Encoding: 8bit\n"
+"Plural-Forms: nplurals=INTEGER; plural=EXPRESSION;\n"
```
Comment (MichaelPFrey) [inline locale/ua.po]: The error report suggests
`"Plural-Forms: nplurals=3; plural=(n%10==1 && n%100!=11 ? 0 : n%10>=2 && n%10<=4 && (n%100<10 || n%100>=20) ? 1 : 2);\n"
`
which roughly matches Russian plural forms
https://github.com/openscad/openscad/blob/e1f043d2d038a2c1839ec39354383b5de2b12cdb/locale/ru.po#L16-L17

File: locale/ua.po [hunk 2]
```diff
@@ -0,0 +1,1446 @@
...
+
+#: src/mainwin.cc:1035
+msgid "Error while compiling '%1'."
+msgstr "Помилка під час компіляції '%1'."
+
+#: src/mainwin.cc:1039
+msgid "Compilation generated %1 warning."
+msgid_plural "Compilation generated %1 warnings."
+msgstr[0] "Компіляція видала %1 попередження."
+msgstr[1] "Компіляція видала %1 попереджень."
```
Comment (MichaelPFrey) [inline locale/ua.po]: I do not speak Ukrainian, but I can suggest you to take Russian as reference for how to implement the multiple plural forms
https://github.com/openscad/openscad/blob/e1f043d2d038a2c1839ec39354383b5de2b12cdb/locale/ru.po#L1167-L1171

File: locale/ua.po (hunk 1 above)
Comment (kromanenko) [reply locale/ua.po]: Thank you for good example! In Ukrainian it should go like this, depending from number:
1 - 4: msgstr[0]
5 - 20: msgstr[1]
21-24: msgstr[0]
25-30: msgstr[1]
31-34: msgstr[0]
35-40: msgstr[1]
...
I think this should work as expected:
"Plural-Forms: nplurals=2; plural=(n%10>=1 && n%10<=4 && (n%100<11 || n%100>14) ? 0"
" : n%10==0 && n%10>=5 && n%10<=9 ? 1);\n"

I am not a programmer, so it's a bit tricky for me :)

**Comment by kromanenko** [comment]
> ./locale/ua.po:9: invalid plural expression
                  Try using the following, valid for Ukrainian:
                  "Plural-Forms: nplurals=3; plural=(n%10==1 && n%100!=11 ? 0 : n%10>=2 && n%10<=4 && (n%100<10 || n%100>=20) ? 1 : 2);\n"

@MichaelPFrey Why do compiler forces to use 3 plural forms? 
For Ukrainian language there are two plural forms, dependence from amount number is following:
1 - 4: msgstr[0]
5 - 20: msgstr[1]
21-24: msgstr[0]
25-30: msgstr[1]
31-34: msgstr[0]
35-40: msgstr[1]
....

**Comment by kromanenko** [comment]
Wrote a spreadsheet of plural forms, number and modulo of division by 10 and by 100:
https://gist.github.com/kromanenko/b19d8ed09ca51d78421275340659b713

**Comment by MichaelPFrey** [comment]
Tough question. I can only speak german and english and both only have singular and plural.

I can quote the documentation:

https://www.gnu.org/software/gettext/manual/html_node/Plural-forms.html
> **Three forms, special cases for numbers ending in 1 and 2, 3, 4, except those ending in 1[1-4]**
> The header entry would look like this: 
```
Plural-Forms: nplurals=3; \
    plural=n%10==1 && n%100!=11 ? 0 : \
           n%10>=2 && n%10<=4 && (n%100<10 || n%100>=20) ? 1 : 2;
```
> Languages with this property include:
> **Slavic family**
> Russian, Ukrainian, Belarusian, Serbian, Croatian 

I do not know whether this is correct or not, but either-way, it is enforced by the compiler.

Maybe there is some documentation for gettext in Ukrainian or you can ask other ukrainian translators.

please also note:

https://www.gnu.org/software/gettext/manual/html_node/Usual-Language-Codes.html#Usual-Language-Codes
> A.1 Usual Language Codes
> 
> For the commonly used languages, the ISO 639-1 standard defines two-letter codes. 
> ...
> ‘uk’
> 
>     Ukrainian.

**Comment by kromanenko** [comment]
Changed **ua** to **uk**, i hope it will not make confusion with United Kingdom :)

**Comment by MichaelPFrey** [comment]
I will do some clean up like removing the ua file and cleaning up the header of the uk file. Then it should be fine to merge.

**Comment by kromanenko** [comment]
> I will do some clean up like removing the ua file and cleaning up the header of the uk file. Then it should be fine to merge.

Thank you!

**Comment by kromanenko** [comment]
Merged on top of current master, now CI passes. Somehow changes from branch topics/ru_tr_fix also caught in here.

**Pull request merged**